bin/update_related_case_caches.py

```diff
@@ -20,12 +20,15 @@
 logger.setLevel(logging.INFO)
 
 
-def recursive_update_related_case_caches(node, case, visited_ids=set()):
+def recursive_update_related_case_caches(node, case, visited_ids=None):
```

Comment:
I think the `visited_ids` is used as a mini cache throughout the execution. Not sure if that was intentional but this implementation will have less node_ids in visited each time it is recursed

Reply:
True. I guess this mini cache saves us from infinite recursion in case we have some weird `sample->sample` connections... 🤷‍♂